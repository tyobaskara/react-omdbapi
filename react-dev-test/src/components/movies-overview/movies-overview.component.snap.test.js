import React from 'react';
import renderer from 'react-test-renderer';
import MoviesOverview from './movies-overview.component';

const observe = jest.fn();
const unobserve = jest.fn();

// you can also pass the mock implementation
// to jest.fn as an argument
window.IntersectionObserver = jest.fn(() => ({
  observe,
  unobserve,
}))

jest
  .mock('../form-search/form-search.container', () => 'FormSearchComponent')
  .mock('../card-movie/card-movie.component', () => 'CardMovieComponent')
  .mock('../modal/modal.component', () => 'ModalComponent')
  .mock('../spinner/spinner.component', () => 'SpinnerComponent');

test('MoviesOverview snaptest', () => {
  const props = {
    moviesList: [{
      Title: 'Title',
      Year: 'Year',
      imdbID: 'imdbID',
    }],
    moviesPage: {
      Title: 'Title',
      Year: 'Year',
      imdbID: 'imdbID',
    },
    isLoadingSearch: false, 
    searchError: '',
    setMoviesPage: () => {}
  };

  const component = renderer.create(
    <MoviesOverview {...props} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('MoviesOverview error search snaptest', () => {
  const props = {
    moviesList: [],
    moviesPage: {},
    isLoadingSearch: false, 
    searchError: 'Something went wrong',
    setMoviesPage: () => {}
  };

  const component = renderer.create(
    <MoviesOverview {...props} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('MoviesOverview loading search snaptest', () => {
  const props = {
    moviesList: [],
    moviesPage: {},
    isLoadingSearch: true, 
    searchError: '',
    setMoviesPage: () => {}
  };

  const component = renderer.create(
    <MoviesOverview {...props} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});