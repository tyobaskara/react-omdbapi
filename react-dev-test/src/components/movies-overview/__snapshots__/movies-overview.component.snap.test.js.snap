// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesOverview error search snaptest 1`] = `
<div
  className="movies-overview"
>
  <FormSearchComponent />
  <ul
    className="movies-list"
  />
  <p
    className="error"
  >
    Movies Not Found..
  </p>
</div>
`;

exports[`MoviesOverview loading search snaptest 1`] = `
<div
  className="movies-overview"
>
  <FormSearchComponent />
  <ul
    className="movies-list"
  />
  <SpinnerComponent />
</div>
`;

exports[`MoviesOverview snaptest 1`] = `
<div
  className="movies-overview"
>
  <FormSearchComponent />
  <ul
    className="movies-list"
  >
    <CardMovieComponent
      Title="Title"
      Year="Year"
      imdbID="imdbID"
      onClick={[Function]}
    />
  </ul>
</div>
`;
